core/primitives/src/block.rs

```diff
@@ -430,6 +507,71 @@ impl Block {
     }
 }
 
+pub enum ChunksCollection<'a> {
```

Comment:
isn't this simply `Cow<'a, Vec<ShardChunkHeader>>`?

Reply:
No, it isn't. The problem I was trying to solve here is unifying the return type for the `chunk` method on `Block`. What I wanted to say is the return type is something like `&Vec<ShardChunkHeader>`, but the issue is that `BlockV1` only has `ShardChunkHeaderV1`, so I need to map over the collection to lift it into the versioned enum. This creates a new data structure locally in the function call, so you cannot return a reference to it. The purpose of `ChunksCollection` is to hold the mapped `Vec` in the `V1` case and simply the reference to the data in the `V2` case.

I also thought about trying to make `chunks` return an iterator instead, but there are lots of places where the result is indexed into at the call-site, so an indexed collection was needed.